### DISCUSSION TITLE
Newbie questions about datasets

### DISCUSSION THREAD

**Discussion by carloscastellanos** [acids-ircam/RAVE#283] [Q&A]
Hey gang, I am trying train RAVE on my own audio samples. I have experience training GANs & VAEs on image data but this is my first time working with VAEs and audio. So a few questions:

- What is the recommended length of audio samples (5 secs, 20 secs, etc?)
- Should all samples in a training set be the same length and sample rate (kind of like how most image-based models require specific dimensions, aspect ratios, etc)?
- How many samples/files is considered a good amount for training (100, 500, 10k?)
- Do I need to do any preparation or preprocessing of the audio files beforehand or does the RAVE preprocess command take care of all preprocessing

Thanks

**Comment by spwilson2** [comment]
> Should all samples in a training set be the same length and sample rate (kind of like how most image-based models require specific dimensions, aspect ratios, etc)?
> Do I need to do any preparation or preprocessing of the audio files beforehand or does the RAVE preprocess command take care of all preprocessing
> What is the recommended length of audio samples (5 secs, 20 secs, etc?)

Length of samples used to train are determined by `seconds = n_signal / sample rate`, so basically you'll want your training data to be >2.5 seconds per sample.

```
  --n_signal: Number of audio samples to use during training
    (default: '131072')
```

Provided samples are automatically converted through preprocess command and chopped up so no extra pre-processing is necessary.
